src/actions/index.js

```diff
@@ -136,24 +113,24 @@ export const renameFolder = (newName, id) => ({
   name: newName
 })
 
-export const createFolder = (name, tempId) => {
+export const createFolder = name => {
   return async (dispatch, getState) => {
-    let existingFolder = getState().files.find(f => f.id !== tempId && f.type === 'directory' && f.name === name)
+    let existingFolder = getState().files.find(f => f.type === 'directory' && f.name === name)
 
     if (existingFolder) {
-      return dispatch({
+      dispatch({
         type: CREATE_FOLDER_FAILURE_DUPLICATE,
-        id: tempId,
         alert: {
           message: 'alert.folder_name',
           messageData: { folderName: name }
         }
       })
+      throw new Error('alert.folder_name')
```

Comment:
Throwing here allows to catch() the error instead of having a single generic then()